### PR TITLE
#913 Added networkType enum. Implemented logic based on network type

### PR DESCRIPTION
**Pull request by artemjackson**
@andrew-aladev please take a look at [this](https://github.com/thaliproject/Thali_CordovaPlugin/pull/919/commits/79086ef51ea6db4c8b0ceb19989d1d9094ff7b1c) implementation.
Does it look well or not?

---
This change is [Reviewable](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/919)

### CODE REVIEW

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (79086ef)

**Comment by ThaliTester** [comment]
Test (Success) 82203060 build is completed (79086ef)

See https://github.com/ThaliTester/TestResults/tree/8220306079086ef__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
Test 8220306079086ef(79086ef) has failed

See https://github.com/ThaliTester/TestResults/tree/8220306079086ef__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the fail logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (2c9dbcd)

**Comment by ThaliTester** [comment]
Test (Success) 82203060 build is completed (2c9dbcd)

See https://github.com/ThaliTester/TestResults/tree/822030602c9dbcd__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
Test 822030602c9dbcd(2c9dbcd) has successfully finished without an error

See https://github.com/ThaliTester/TestResults/tree/822030602c9dbcd__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (708b260)

**Comment by ThaliTester** [comment]
Test (Fail) 82203060 build is completed (708b260)

See https://github.com/ThaliTester/TestResults/tree/82203060708b260__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

File: thali/NextGeneration/thaliMobile.js [hunk 1]
```diff
@@ -30,11 +30,76 @@ var getCombinedResult = function (results) {
...
+ * @enum {string}
+ */
+var networkTypes = {
+  WIFI: 'WIFI',
+  NATIVE: 'NATIVE',
+  BOTH: 'BOTH'
+};
+module.exports.networkTypes = networkTypes;
+
+var getWifiMethodByNetworkType = function (method, networkType) {
```
Comment (andrew-aladev) [inline thali/NextGeneration/thaliMobile.js]: I think we can have a good function like `getMethodByNetworkType (target, method, networkType)` where target equals `thaliWifiInfrastructure` or `ThaliMobileNativeWrapper`.

File: thali/NextGeneration/thaliMobile.js [hunk 2]
```diff
@@ -142,37 +207,33 @@ var handleNetworkChanged = function (networkChangedValue) {
...
-        thaliWifiInfrastructure.start(router, pskIdToSecret)
-      ),
-      promiseResultSuccessOrFailure(
-        ThaliMobileNativeWrapper.start(router, pskIdToSecret)
-      )
-    ]).then(function (results) {
-      resolve(getCombinedResult(results));
-    });
+
+    start = getWifiOrNativeMethodByNetworkType('start',
```
Comment (andrew-aladev) [inline thali/NextGeneration/thaliMobile.js]: Is `start` a global variable? Maybe we can have `var start` here.

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (884b823)

**Comment by andrew-aladev** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/919#-:-KPqvYTzZnRj8afn00_k:bnfp4nl)*

**Comment by ThaliTester** [comment]
Test (Success) 82203060 build is completed (884b823)

See https://github.com/ThaliTester/TestResults/tree/82203060884b823__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
Test 82203060884b823(884b823) has successfully finished without an error

See https://github.com/ThaliTester/TestResults/tree/82203060884b823__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by yaronyg** [comment]
Once you fix the switch  issues (and do whatever you want with the NB issue) then :lgtm: 

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[thali/NextGeneration/thaliMobile.js, line 62 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/919#-KPsm8dllKX7s1LAGz4N:-KPsm8dllKX7s1LAGz4O:beibjpo) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/884b8234eef404973d02b09189c1770411a32e4c/thali/NextGeneration/thaliMobile.js#L62)):*
> ```JavaScript
> 
> var getWifiOrNativeMethodByNetworkType = function (method, networkType) {
>   if (networkType === networkTypes.BOTH) {
> ```

Please use a switch statement

---

*[thali/NextGeneration/thaliMobile.js, line 220 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/919#-KPsn_KP0mQd_0868OVn:-KPsn_KQImSvMZonvJ_b:biekwou) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/884b8234eef404973d02b09189c1770411a32e4c/thali/NextGeneration/thaliMobile.js#L220)):*
> ```JavaScript
>     var start = getWifiOrNativeMethodByNetworkType('start',
>      thaliMobileStates.networkType);
>     start(router, pskIdToSecret)
> ```

NB: I find this code a bit surprising.  I probably would have done something like getWifiOrNativeMethodByNetworkType('start', thaliMobileStates.networkType)(router, pskIdToSecret). The fact that getWifiOrNativeMethodByNetworkType returns a function rather than a promise was a bit surprising (but understandable) to me. It got in the way of reading the code. By putting the args right next to the function output it's more immediately clear what's going on. In any case, I'm arguing over nits.

---

*[thali/NextGeneration/thaliMobile.js, line 386 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/919#-KPsonPE_bKQYW_ohxD2:-KPsonPFnqbN5xIWHEyt:b-n7chf0) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/884b8234eef404973d02b09189c1770411a32e4c/thali/NextGeneration/thaliMobile.js#L386)):*
> ```JavaScript
>   var networkType = thaliMobileStates.networkType;
>   return promiseQueue.enqueue(function (resolve, reject) {
>     if (networkType === networkTypes.NATIVE ||
> ```

Again, please use Switch

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/919#-:-KPsowv7vJN4RrD3HR8c:b74q3zg)*

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (198550b)

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (a181f1c)

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 3. task. (5965752)

**Comment by ThaliTester** [comment]
Test (Success) 82203060 build is completed (198550b)

See https://github.com/ThaliTester/TestResults/tree/82203060198550b__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
Test (Fail) 82203060 build is completed (a181f1c)

See https://github.com/ThaliTester/TestResults/tree/82203060a181f1c__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
Test 82203060198550b(198550b) has failed

See https://github.com/ThaliTester/TestResults/tree/82203060198550b__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the fail logs

**Comment by ThaliTester** [comment]
Test (Success) 82203060 build is completed (5965752)

See https://github.com/ThaliTester/TestResults/tree/822030605965752__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
Test 822030605965752(5965752) has successfully finished without an error

See https://github.com/ThaliTester/TestResults/tree/822030605965752__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 6. task. (b7ba7ed)

**Comment by ThaliTester** [comment]
Test (Fail) 82203060 build is completed (b7ba7ed)

See https://github.com/ThaliTester/TestResults/tree/82203060b7ba7ed__913_Added_networkType_enum__Implemented_logic_based_on_network_type_artemjackson/ for the logs

**Pull request closed without merging**